collectors/modality-probe-collector-common/src/lib.rs

```diff
@@ -93,17 +99,70 @@ pub struct Report {
     pub probe_clock: LogicalClock,
     pub seq_num: SequenceNumber,
     pub persistent_epoch_counting: bool,
+    pub time_resolution: NanosecondResolution,
+    pub wall_clock_id: WallClockId,
     pub frontier_clocks: Vec<LogicalClock>,
     pub event_log: Vec<EventLogEntry>,
 }
 
 #[derive(Debug, Eq, PartialEq, Clone, Copy)]
 pub enum EventLogEntry {
-    Event(EventId),
-    EventWithPayload(EventId, u32),
-    TraceClock(LogicalClock),
+    Event(Option<Nanoseconds>, EventId),
```

Comment:
This is optional, and stylistic, but I would softly recommend flattening out the Option<Nanoseconds> by adding 3 new variants instead `TimedEvent`, `TimedEventWithPayload`, and `TimedTraceClock`.  It would reduce code motion around existing handling of specific variants and firmly point to more direct un-nested handling of each entry.

Reply:
Yeah I think that makes sense too, will get this updated.